plugins/elasticsearch-bboss/src/main/java/com/navercorp/pinpoint/plugin/elasticsearchbboss/interceptor/ElasticsearchExecutorOperationInterceptor.java

```diff
@@ -0,0 +1,187 @@
+        super(context, descriptor);
+        recordResult = this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordResult",false);
+        recordArgs = this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordArgs",true);
+        recordDsl =  this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordDsl",true);
+        maxDslSize =  this.getTraceContext().getProfilerConfig().readInt("profiler.elasticsearchbboss.maxDslSize",ElasticsearchConstants.maxDslSize);
+        recordResponseHandler =  this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordResponseHandlerClass",false);
+        recordESVersion = this.getTraceContext().getProfilerConfig().readBoolean("profiler.elasticsearchbboss.recordESVersion",true);
+
+
+    }
+    @Override
+    public void before(Object target, Object[] args) {
+        super.before(target,args);
+
+    }
+    @Override
+    public void after(Object target, Object[] args, Object result, Throwable throwable){
+
+
+        super.after(target,args,result,throwable);
+    }
+
+
+    @Override
+    protected void doInBeforeTrace(SpanEventRecorder recorder, Object target, Object[] args) {
+
+        recorder.recordServiceType(ElasticsearchConstants.ELASTICSEARCH_EXECUTOR);
+    }
+
+    private String getClusterVersionInfo(Object target){
+        try {
+            if(getClusterVersionInfo != null)
+                return (String)getClusterVersionInfo.invoke(target);
+            synchronized (this) {
+                if(getClusterVersionInfo == null) {
+                    Method _getClusterVersionInfo = target.getClass().getMethod("getClusterVersionInfo");
+                    getClusterVersionInfo = _getClusterVersionInfo;
+                }
+            }
+            return (String)getClusterVersionInfo.invoke(target);
+        } catch (NoSuchMethodException e) {
+
+        } catch (IllegalAccessException e) {
+
+        } catch (InvocationTargetException e) {
+
+        }
+        return "";
+
+    }
+    private String getEndPoint(Object[] args){
+        String url = (String)args[0];
+        //http://xxx:9200/
+        int idx = url.indexOf("://");
+        String endPoint = null;
+        if(idx > 0){
+            int sub = url.indexOf('/',idx + 3);
+            if(sub > 0 ){
+                endPoint = url.substring(idx + 3,sub);
+            }
+            else
+            {
+                endPoint = url.substring(idx + 3);
+            }
+        }
+        else{
+            endPoint = "Unknown";
+        }
+        return endPoint;
+
+    }
+    @Override
+    protected void doInAfterTrace(SpanEventRecorder recorder, Object target, Object[] args, Object result,
+            Throwable throwable) {
+        String elasticsearchClusterVersionInfo = getClusterVersionInfo( target);
```

Comment:
I think retrieving cluster version can go under `if(recordESVersion)` scope.
Since `elasticsearchClusterVersionInfo` variable is only used there.

Reply:
Modify code as follow
```java
recorder.recordApi(getMethodDescriptor());

        recorder.recordDestinationId("ElasticsearchBBoss");

        recorder.recordEndPoint(getEndPoint( args));
        if(recordESVersion) {
            // Each target(ClientInstance) has a specific version of Elasticsearch Datasource,
            // and each Elasticsearch Datasource retains its corresponding Elasticsearch cluster version information
            // such as Elasticsearch 1.x or 2.x or 5.x or 6.x or 7.x or 8.x and so on.
            // so we should get elasticsearchClusterVersionInfo in target everytime.
            String elasticsearchClusterVersionInfo = getClusterVersionInfo( target);
            recorder.recordAttribute(ElasticsearchConstants.ARGS_VERSION_ANNOTATION_KEY, elasticsearchClusterVersionInfo);//record elasticsearch version and cluster name.
```